core/mm/vm.c

```diff
@@ -266,7 +266,7 @@ TEE_Result vm_map_pad(struct user_mode_ctx *uctx, vaddr_t *va, size_t len,
 	struct vm_region *reg = NULL;
 	uint32_t attr = 0;
 
-	if (prot & ~TEE_MATTR_PROT_MASK)
+	if (prot & ~TEE_MATTR_PROT_MASK && prot & ~TEE_MATTR_DEVICE_PROT_MASK)
```

Comment:
This is not the type of check I expect that this function should perform.

Reply:
Do you want me to just remove:
`if (prot & ~TEE_MATTR_PROT_MASK)`
Or do you have anything else specifically in mind?

Reply:
Please remove `&& prot & ~TEE_MATTR_DEVICE_PROT_MASK`.

Reply:
The reason I add the check for the `TEE_MATTR_DEVICE_PROT_MASK` is that for device regions we have some extra flags that are not inside the `TEE_MATTR_PROT_MASK`. I thought having two different flags on for testing the device regions and one for testing the normal region would be the best approach.  Since the normal regions  don't have the TEE_MATTR_DEVICE_nGnRE and TEE_MATTR_DEVICE_nGnRnE flags.
Do you want me to update the `TEE_MATTR_PROT_MASK` flag instead to also cover the device regions?

Reply:
The flags you're mentioning aren't protection or permission flags. I'd expect device mapping attributes to  come from the `mobj` describing the memory to map. Check how `attr` is constructed below.

Reply:
Ack